pkg/apis/navigator/types.go

```diff
@@ -151,12 +153,51 @@ type NavigatorClusterConfig struct {
 	PilotImage ImageSpec
 
 	SecurityContext NavigatorSecurityContext
+
+	Paused bool
+}
+
+type NavigatorClusterStatus struct {
+	Conditions []ClusterCondition
 }
 
 type NavigatorSecurityContext struct {
 	RunAsUser *int64
 }
 
+type ClusterConditionType string
+
+const (
+	// Available means the cluster is available, ie. at least the minimum available
+	// replicas required are up and running for at least minReadySeconds.
+	ClusterAvailable ClusterConditionType = "Available"
+	// Progressing means the cluster is progressing. Progress for a deployment is
+	// considered when a new replica set is created or adopted, and when new pods scale
+	// up or old pods scale down. Progress is not estimated for paused deployments or
+	// when progressDeadlineSeconds is not specified.
+	ClusterProgressing ClusterConditionType = "Progressing"
+	// ReplicaFailure is added in a cluster when one of its pods fails to be created
+	// or deleted.
+	ClusterReplicaFailure ClusterConditionType = "ReplicaFailure"
```

Comment:
I can't see where we actually use/set these conditions?

Also, update comments (assuming they've come from Deployment defs) and remove conditions we don't currently implement.

Reply:
Actually implemented now! (I should have added wip back to the title)